includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1479,13 +1480,35 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
 		$charge_id = $order->get_meta( '_charge_id', true );
 
 		try {
-			if ( is_null( $amount ) ) {
-				// If amount is null, the default is the entire charge.
-				$refund = $this->payments_api_client->refund_charge( $charge_id );
+			// If the payment method is Interac_Present, the refund should already have been done in the mobile app, so we only record the refund without actioning it.
+			if ( Payment_Method::INTERAC_PRESENT === $this->get_payment_method_type_for_order( $order ) ) {
+				$refunds = $this->payments_api_client->list_refunds( $charge_id )['data'];
+
+				$expected_refund_amount        = $amount ?? $order->get_total();
+				$stripe_expected_refund_amount = WC_Payments_Utils::prepare_amount( $expected_refund_amount, $order->get_currency() );
+
+				$filtered_refunds = array_filter(
+					$refunds,
+					function ( $r ) use ( $stripe_expected_refund_amount ) {
+						return 'succeeded' === $r['status'] && $stripe_expected_refund_amount === $r['amount'];
+					}
+				);
+
+				if ( isset( $filtered_refunds[0] ) ) {
+					$refund = $filtered_refunds[0];
+				} else {
+					$interac_refund_error = __( 'Interac in-person payments must be refunded using the mobile app, with the customer present.', 'woocommerce-payments' );
+					return new WP_Error( 'wcpay_edit_order_interac_present_refund_failure', $interac_refund_error, [ 'status' => 404 ] );
+				}
```

Comment:
Minor: Reverses our check that a refund exists, which arguably makes this easier to read.

```suggestion
				if ( ! isset( $filtered_refunds[0] ) ) {
					$interac_refund_error = __( 'Interac in-person payments must be refunded using the mobile app, with the customer present.', 'woocommerce-payments' );
					return new WP_Error( 'wcpay_edit_order_interac_present_refund_failure', $interac_refund_error, [ 'status' => 404 ] );
				}
				
				$refund = $filtered_refunds[0];
```